File: src/components/hull.rs
pub struct Hull {
    max_hull: u8,
    current_hull: u8,
    repair_rate: u8,
}

impl Default for Hull {
    fn default() -> Self {
        Self {
            max_hull: 100,
            current_hull: 100,
            repair_rate: 5,
        }
    }
}

File: StarTrekExplorers/Components/Ship/Capabilities/Hull.cs
using StarTrekExplorers.Components.Interfaces;

namespace StarTrekExplorersTests.Entities
{
    public class Hull : IDefense
    {
        public int Current { get; private set; } = 100;
        public int Maximum { get; } = 100;
        public int RepairRate { get; } = 5;

        public void TakeDamage(int damage)
        {
            if (damage >= Current)
            {
                Current = 0;
            }
            else
            {
                Current -= damage;
            }
        }
    }
}